compiler/annotation/mod.rs

```diff
@@ -168,6 +170,11 @@ typedb_error!(
             label: String
         ),
         ValueTypeNotFound(8, "Value type '{name}' was not found.", name: String),
+        AnnotationsUnavailableForVariableInInsert(9,
```

Comment:
@krishnangovindraj  notice how every message here doesn't use `Annotations` but instead uses `Types` - `TypesUnavailableForVariableInInsert` would be more consistent